### DISCUSSION TITLE
How to create a new version of your repository on github

### DISCUSSION THREAD

**Discussion by gouneken** [community/community#24613] [Apps, API and Webhooks]
<p>good evening to the whole community, I’m new to Github. I recently created a repository on github, I’m looking to save the changes I made on my computer on github and without having to change the existing repository so that if I have a code error, I can go back to the previous version of my app. But I don’t even know how to even modify my repository. Github’s tutorial shows how to edit a file but not the entire project especially since there are several new folders that have been added to it. when I run the command <code>git push</code> at the repository level to be modified but I got the answer</p>
<blockquote>
<p>already up-to-date</p>
</blockquote>
<p>yet when I look at my git repository, nothing has been updated. Also I ask for help from more experienced members of the community to help me solve a problem on how to create a new version of its repository on git. Thank you, I look forward to your help!</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group quote-modified" data-username="gouneken" data-post="1" data-topic="253939">
<div class="title">
<div class="quote-controls"></div>
 gouneken:</div>
<blockquote>
<p>when I run the command <code>git push</code> at the repository level to be modified but I got the answer</p>
<blockquote>
<p>already up-to-date</p>
</blockquote>
<p>yet when I look at my git repository, nothing has been updated.</p>
</blockquote>
</aside>
<p>This sounds like you didn’t <em>commit</em> your changes yet. Essentially each commit is a version of your repository, and <code>git push</code> transfers new commits to the remote repository.</p>
<p>I highly recommend you take a look at the <a href="https://git-scm.com/book/en/v2" class="inline-onebox">Git - Book</a>. Chapter 2 covers the basics, including the answers to your question: <a href="https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository">How to record changes</a>, and <a href="https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes">exchange them with remote repositories</a>.</p>

**Comment by jfullstackdev** [comment]
<p>there is no public repo too, it’s hard to guess it,</p>
<p>maybe you mean from time to time there is an update, you will be creating a release, because that’s the latest version</p>

**Comment by gouneken** [comment]
<p>Oui, c’est exactement ce que je veux dire et je ne sais pas comment faire.</p>

**Comment by Johnvan** [comment]
I am absolutely lost in the use of this GitHub thing, I have a project connected to GitHub and I cannot find that code, let alone, download a previous version.

**Comment by Johnvan** [comment]
You've got this word "add" around your website and even here "Got remote add". This whole system uses words that mean nothing or are confusing. I think your system is trying to do too much, and even GitHub is confused. Please spend more time in reorganising your system and documentation, covering each subject in full and more description. I used to be a Technical Writer and I can't see how things work in GitHub.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by airtower-luna** [reply]
As I mentioned above, check the [Git Book](https://git-scm.com/book/en/v2) (chapters 1-3 cover the basics). GitHub is built around Git, if you don't understand Git it's bound to be confusing.
